### PR TITLE
Update log extension

### PR DESCRIPTION
**Pull request by bwheelz36**
This is a fix for #325. For a small issue it turned out to be quite a lot of work!

This is quite a "hard" change: any time logs are written, the code will always use the "log" extension, regardless of what the user entered. 
Reading the logs functions as before, which means the code will still be able to read in old log files.

### CODE REVIEW

**Comment by bwheelz36** [comment]
I can't really understand this failed codecov test...

**Comment by bwheelz36** [comment]
you are probably right, forcing it is probably over the top.

so you think we could just leave it to the user to handle the extension? or perhaps put a keyword variable in the `JSONLogger.__init__` `extension=log`?

**Comment by till-m** [comment]
> so you think we could just leave it to the user to handle the extension?

In my opinion, yes. I think most storing API's in python handle it like that, e.g. `np.savetxt` or `pickle`.

IMO, the best thing would be to make all our examples/documentation use `.log`, but otherwise let the user decide based on the input string.

**Comment by bwheelz36** [comment]
Ok, I've made it now so it will simply call the log file exactly what the user calls it, and (I think) I've updated all docs such that .log extension is used...

File: bayes_opt/logger.py [hunk 1]
```diff
@@ -135,7 +136,9 @@ def update(self, event, instance):
 
 class JSONLogger(_Tracker):
```
[the remark this answers is not in the document]
Comment (bwheelz36) [reply bayes_opt/logger.py:137]: yeahhhhhhh. It's a bit discordant. But I think overall we are probably better to leave it as is? after, each entry IN the logs is json...

**Pull request merged**